tests/mlflow/tensorflow/test_tensorflow_model_export.py

```diff
@@ -15,38 +16,59 @@
-        self._sess = tf.Session()
-        # Training and creating expected predictions on training dataset.
-        self._dnn.train(self._input_train, steps=100)
-        self._dnn_predict = self._dnn.predict(self._input_train)
+
+    def helper(self, feature_spec, tmp, estimator, df):
+        """
+        This functions handles exporting, logging, loading back, and predicting on an estimator for 
+        testing purposes.
+        """
+        receiver_fn = tf.estimator.export.build_raw_serving_input_receiver_fn(feature_spec)
+        saved_estimator_path = tmp.path("model")
+        os.makedirs(saved_estimator_path)
+        # Saving Tensorflow model.
+        saved_estimator_path = estimator.export_savedmodel(saved_estimator_path, 
+                                                       receiver_fn).decode("utf-8")
+        # Logging the Tensorflow model just saved.
+        tensorflow.log_saved_model(saved_model_dir=saved_estimator_path,
+                                   signature_def_key="predict", 
+                                   artifact_path=tmp.path("hello"))
+        # Loading the saved Tensorflow model as a pyfunc.
+        x = pyfunc.load_pyfunc(saved_estimator_path)
+        # Predicting on the dataset using the pyfunc.
+        xpred = x.predict(df)
+        return [row for _, row in xpred.iterrows()]
 
     def test_log_saved_model(self):
+        # This tests model logging capabilities on the sklearn.iris dataset.
         with TempDir(chdr=False, remove_on_exit=True) as tmp:
+            iris = datasets.load_iris()
+            X = iris.data[:, :2]  # we only take the first two features.
+            y = iris.target
+            trainingFeatures = {}
+            feature_names = iris.feature_names[:2]
+            for i in range(0, 2):
+                # Tensorflow is fickle about feature names, so we remove offending characters
+                iris.feature_names[i] = iris.feature_names[i].replace(" ", "")
+                iris.feature_names[i] = iris.feature_names[i].replace("(", "")
+                iris.feature_names[i] = iris.feature_names[i].replace(")", "")
+                trainingFeatures[iris.feature_names[i]] = iris.data[:, i:i+1]
+            tf_feat_cols = []
+            feature_names = iris.feature_names[:2]
+            # Creating Tensorflow-specific numeric columns for input.
+            for col in iris.feature_names[:2]:
+                tf_feat_cols.append(tf.feature_column.numeric_column(col))
+            # Creating input training function.
+            input_train = tf.estimator.inputs.numpy_input_fn(trainingFeatures, 
+                                                                        y, 
+                                                                        shuffle=False, 
+                                                                        batch_size=1)
+            # Creating Deep Neural Network Regressor. 
+            dnn = tf.estimator.DNNRegressor(feature_columns=tf_feat_cols, 
```

Comment:
Maybe rename `dnn` -> `estimator`, `dnn_predict` -> `estimator_preds` so it's easier to see that the code path here is similar to the one in the other test